arangod/Aql/ExecutionBlockImpl.tpp

```diff
@@ -349,6 +349,25 @@ ExecutionBlockImpl<Executor>::ExecutionBlockImpl(
 
 template<class Executor>
 ExecutionBlockImpl<Executor>::~ExecutionBlockImpl() {
+  // Double use diagnostics:
+  uint64_t userCount = _numberOfUsers.fetch_add(1);
+  if (userCount > 0) {
+    LOG_TOPIC("52637", WARN, Logger::QUERIES)
+        << "ALERT: Double use of ExecutionBlock detected, stacktrace:";
+    CrashHandler::logBacktrace();
+    _logStacktrace.store(true, std::memory_order_relaxed);
+  }
+  auto guard = scopeGuard([&]() noexcept {
+    uint64_t userCount = _numberOfUsers.fetch_sub(1);
+    if (_logStacktrace.load(std::memory_order_relaxed)) {
+      LOG_TOPIC("52638", WARN, Logger::QUERIES)
+          << "ALERT: Found _logStacktrace:";
+      CrashHandler::logBacktrace();
+      if (userCount == 0) {
+        _logStacktrace.store(false, std::memory_order_relaxed);
+      }
+    }
+  });
   stopAsyncTasks();
```

Comment:
We must also do the user-accounting when calling `stopAsyncTasks` directly from `~ExecutionEngine` (before `~ExecutionBlockImpl` destructors are executed).

Reply:
Totally right, will fix.

Reply:
Moved user checking to `stopAsyncTasks()`.